components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java

```diff
@@ -1391,6 +1392,35 @@ public static String appendQueryParamsStringToUrl(String url, String queryParamS
         return queryAppendedUrl;
     }
 
+    /**
+     * Append a query param map to the URL (URL may already contain query params)
+     *
+     * @param url         URL string to append the params.
+     * @param queryParams Map of query params to be append.
+     * @return Built URL with query params.
+     * @throws UnsupportedEncodingException Throws when trying to encode the query params.
+     */
+    public static String appendQueryParamsToUrl(String url, Map<String, String> queryParams)
+            throws UnsupportedEncodingException {
+
+        if (StringUtils.isEmpty(url)) {
+            throw new IllegalArgumentException("Passed URL is empty.");
+        }
+        if (queryParams == null) {
```

Comment:
Is this correct?
Why can't we handle this gracefully like we handle an empty map?

Reply:
Deprecate the existing and introduce a new method.

Reply:
PR sent https://github.com/wso2/carbon-identity-framework/pull/2012